File: Placement Sheet/Stack & Queue/NextLargerElement.java
import java.util.Stack;

public class NextLargerElement {
    public static int[] findNextGreaterElm(int[] arr) {
        Stack<Integer> stack = new Stack<>();
        int[] result = new int[arr.length];
        for (int i = 0; i < arr.length; i++) {
            while (!stack.isEmpty() && arr[i] > arr[stack.peek()]) {
                int index = stack.pop();
                result[index] = arr[i];
            }
            stack.push(i);
        }
        while (!stack.isEmpty()) {
            result[stack.pop()] = -1;
        }

        return result;
    }

    public static void display(int[] arr) {
        if (arr.length == 0) {
            System.out.println("Array is empty...");
            return;
        }
        for (int i = 0; i < arr.length; i++) {
            System.out.print(arr[i] + ", ");
        }
    }

    public static void main(String[] args) {
        int[] arr = { 4, 5, 2, 10, 8 };
        int[] result = findNextGreaterElm(arr);
        display(result);
    }
}

File: Placement Sheet/Stack & Queue/NextLargerElement.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var NextLargerElement = /** @class */ (function () {
    function NextLargerElement() {
    }
    NextLargerElement.prototype.findNextGreater = function (arr) {
        var result = new Array(arr.length).fill(-1);
        var stack = [];
        for (var i = 0; i < arr.length; i++) {
            while (stack.length > 0 && arr[i] > arr[stack[stack.length - 1]]) {
                var index = stack.pop();
                result[index] = arr[i];
            }
            stack.push(i);
        }
        return result;
    };
    return NextLargerElement;
}());
var arr = [4, 5, 2, 10, 8];
var nge = new NextLargerElement();
var result = nge.findNextGreater(arr);
console.log(result);
